File: leetcode/solutions/0033.SearchInRotatedSortedArray/search_test.ts
import { test } from "https://deno.land/std/testing/mod.ts";
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import search from "./search.ts";

test("0033. Search in Rotated Sorted Array", () => {
  assertEquals(search([4,5,6,7,0,1,2], 0), 4);
  assertEquals(search([4,5,6,7,0,1,2], 3), -1);
  assertEquals(search([5,1,3], 1), 1);
  assertEquals(search([5,1,3], 3), 2);
});

File: leetcode/solutions/0033.SearchInRotatedSortedArray/search_test.go
package solution

import "testing"

func Test_0033_SearchInRotatedSortedArray(t *testing.T) {
	result := search([]int{4, 5, 6, 7, 0, 1, 2}, 0)
	expected := 4

	if result != expected {
		t.Errorf("TEST FAILED! got: %#v, want: %#v.", result, expected)
	}

	result = search([]int{4, 5, 6, 7, 0, 1, 2}, 3)
	expected = -1

	if result != expected {
		t.Errorf("TEST FAILED! got: %#v, want: %#v.", result, expected)
	}
}
